nexus/reconfigurator/planning/src/blueprint_builder/builder.rs

```diff
@@ -2342,7 +2332,15 @@ pub mod test {
             .expect("failed to create builder");
 
             assert!(builder.disks.changed_disks.is_empty());
-            assert!(builder.disks.parent_disks.is_empty());
+            // We expect sleds to be present but not have any disks in them.
+            for (sled_id, disks) in builder.disks.parent_disks {
+                assert_eq!(
+                    disks.disks,
+                    Vec::new(),
+                    "for sled {}, no disks present in parent",
```

Comment:
nitty suggested:

```suggestion
                    "for sled {}, expected no disks present in parent, but found some",
```

Actually, would you get an even more informative message with `assert_eq!(builder.disks.parent_disks, BTreeMap::new())`?